chapters/03-package-setup.Rmd

```diff
@@ -1,37 +1,55 @@
-# Setting up an R package
+# Setting up an R package {#package-setup}
 
 - Describing what a package is and when or why to make one
+    - Use [available](https://github.com/r-lib/available) to make a name (naming is hard).
 - Using devtools, usethis for development
-- e.g. `use_devtools()` and `use_usethis()`.
+    - e.g. `use_devtools()` and `use_usethis()`.
 - Developing functions
 - Basic explanation of what a function is and its components
 - Actual process of taking code and converting it into a function will
     be done in next chapter
 - Making and using datasets
 - Downloading zipf data at this point, using `usethis::use_data_raw()`, write the script to download the data and process into `data/` via `usethis::use_data()`.
 
+## About Zipf's Law
 
+We'll create a package (`zipfs`) to analyze a fascinating result in the field of
+quantitative linguistics. Zipf’s Law states that the second most common word in
+a body of text appears half as often as the most common, the third most common
+appears a third as often, and so on. To test whether Zipf’s Law holds for a
+collection of classic novels that are freely available from Project Gutenberg,
+we write a software package that counts and analyzes the word frequency
+distribution in any arbitrary body of text.
 
-## Final exercise
+## Final exercise {#03-final-exercise}
 
-<!-- NOTE: Preferably something linked to working towards completing the project assignments, maybe to be used as a milestone? -->
+Complete these tasks as part of the package for your first project assignment.
+Refer to the project assignment for exact details on what you are required to
+create and on what the package should do. For this exercise, we will be using
+an package about weather in Kenya as an example and reference.
 
-1. Use `create_package()` to create a package called FIXME.  
+1. Decide on a simple but descriptive name for your new package. Use
```

Comment:
Do we need to provide additional guidance here, since this is potentially being assigned to many classes of students?

Reply:
Yea, I'm not 100% certain how to do that. Let's add a todo to the for instructor section about it.